src/main/scala/viper/server/vsi/VerificationServer.scala

```diff
@@ -323,14 +320,20 @@ abstract class VerificationTask()(implicit val executionContext: ExecutionContex
   protected def enqueueMessages(msg: A): Unit = {
     implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
 
-    var current_offer: Future[QueueOfferResult] = null
     val answer = q_actor ? TaskProtocol.BackendReport(pack(msg))
-    current_offer = answer.flatMap({
-      case res: Future[QueueOfferResult] => res
-    })
-    while(current_offer == null || !current_offer.isCompleted){
-      Thread.sleep(10)
+    Await.ready(answer, Duration.Inf)
```

Comment:
This part would need to be adapted after PR22 is merged. I suspect that rather than waiting for the future to end we should just return it to the client. Working with futures is quite convenient because they can always be flattened, but flattening (and, moreover, await) should happen lazily.

Reply:
I agree that futures should be passed on the clients instead of awaiting them in ViperServer. However, I do not agree with you in this particular use since this function is concerned with enqueuing messages to the message queue. As we've discussed in our meetings, the verification job should be slowed down (i.e. blocked) when the queue is full. Or did I miss something?

Reply:
Okay, but then, I think, we need to Await only once, in the end of ```enqueueMessages```. 
How about something like this?: 

```scala
protected def enqueueMessages(msg: A): Unit = {
  implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
  val answer = q_actor ? TaskProtocol.BackendReport(pack(msg))
  val current_offer = answer.map {
    case Success(current_offer) => //...
    // ... 
  }
  Await.result(current_offer, Duration.Inf)
}
```

Reply:
I changed it now to (newly located in `MessageStreamingTask`):
```
// answer is a future that will resolve with the actor's response to the BackendReport request
val answer = (q_actor ? TaskProtocol.BackendReport(msg)).mapTo[Future[QueueOfferResult]]
// currentOffer is the future that the actor will send in its response (assuming that no timeout occurred requesting it from the actor)
// currentOffer will resolve when the message is dequeued from the queue
val currentOffer = answer.flatten
try {
  // note that an exception is thrown if the currentOffer future fails, e.g. because the askTimeout occurred
  Await.result(currentOffer, Duration.Inf)
} catch {
  case ex: Exception =>
    // print exception such that one sees that something went wrong:
    println(s"exception in enqueueMessage occurred: $ex")
    // rethrow exception:
    throw ex
}
```

Reply:
Generally looks reasonable now. As a recommendation for future, ```Duration.Inf``` should be avoided in favour of fault tolerance (maybe worth adding a TODO).